scripts/fix_sq_headers.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+"""
+Usage: fix_sq_headers list|update
+
+Uses the usual config entries (workflow.dataset/.access_level) and:
+
+  images.samtools      Path of an image containing samtools
+  references.expected  Name (from KNOWN_REFS) of the (bad) refset expected
+  references.new       Dict file containing the replacement @SQ headers
+"""
+
+import hashlib
+import os
+import subprocess as sb
+import sys
+
+from hailtop.batch.resource import JobResourceFile
+
+from cpg_utils import Path, to_path
+from cpg_utils.hail_batch import get_batch, get_config, image_path
+from metamist.graphql import query, gql
+
+FIND_CRAMS = gql("""
+query CramQuery($project: String!) {
+  project(name: $project) {
+    sequencingGroups {
+      id
+      analyses(type: {eq: "cram"}) {
+        id
+        meta
+        output
+        active
+        type
+      }
+    }
+  }
+}
+""")
```

Comment:
Just performed this query myself and it outputs files that don't exist. This should be fine because you check the existence of files prior to performing reheadering. But is there a possibility of a file we don't want to reheader but also, don't want to delete? 

I'm wondering if it would be best to pull the latest cram analyses 

So the query would look like this (with `timestampCompleted` included):
```
query CramQuery($project: String!) {
  project(name: $project) {
    sequencingGroups {
      id
      analyses(type: {eq: "cram"}) {
        id
        meta
        output
        active
        type
        timestampCompleted
      }
    }
  }
}
```
And then sort the analyses based on `timestampCompleted`
```
sorted_analyses = sorted(
  analyses,
  key=lambda x: datetime.strptime(
      x['timestampCompleted'], '%Y-%m-%dT%H:%M:%S'
  ),
)
(snippet from Eddie's auditing scripts)
```

Reply:
Actually, this may just capture the NAGIM crams, which if I understand correctly, don't have the issue this script is trying to solve 🙃 Leaving this here in case I'm wrong!

Reply:
The script doesn't do any deleting itself yet, so it's safe from this point of view.

However [running it in test](https://batch.hail.populationgenomics.org.au/batches/430059/jobs/1) does come up with a number of seemingly faulty database records with filenames that aren't CRAM or don't exist, which the script is careful to check for. I am hoping that the data will be cleaner in main, but something like this may be a good idea.

So we might have multiple CRAM analyses for a sequencing group, and we ought to only look at the most recent ones? Actually I guess I should be paying attention to the `active` field too…

Reply:
Rather than look at the timestamps and active field, we've (for now) de-emphasised querying these paths from metamist in favour of just specifying a bucket path on the command line / in the config file.